features/achordion.c

```diff
@@ -139,6 +151,11 @@ bool process_achordion(uint16_t keycode, keyrecord_t* record) {
   }
 
   if (achordion_state == STATE_UNSETTLED && record->event.pressed) {
+#if ACHORDION_TYPING_STREAK_TIMEOUT > 0
+    const bool is_streak = timer_elapsed(idle_timer) < ACHORDION_TYPING_STREAK_TIMEOUT;
```

Comment:
A limitation with 16-bit timers is that they wrap around every 65.536 seconds. I suppose it's a rare occurrence in practice, but it looks conceivable to have a "false positive streak" where `is_streak` could be incorrectly true with a sequence like (1) tap a non-tap-hold key, (2) wait slightly less than 65.536 seconds, (3) holding a tap-hold key, (4) pressing another key. There might be similar pathologies when the keyboard first boots up where the timer is initially 0.

Here is a potential approach to avoid timer wraparound:

1. Consider a zero-valued timer to represent that it has expired.
2. Reset the timer like this: 
   ```c
   streak_timer = (timer_read() + ACHORDION_STREAK_TIMEOUT) | 1;
   ```
   The bitwise-or with 1 ensures a nonzero value to distinguish it from "expired" state. This comes at the cost of effectively coarsening time resolution to steps of 2 ms, but that's still good for tracking key events.
3. In `achordion_task()`, do 

   ```c
   if (streak_timer && timer_expired(timer_read(), streak_timer)) {
     streak_timer = 0;  // Expired.
   }
   ```
4. Here at this point of the code, check whether the timer is active with
  
   ```c
   const bool is_streak = (streak_timer != 0);
   ```

Reply:
That is great, thank you for the explanation. I was also worried about this issue, and I thought that using `timer_expired` would take care of that. I'll update the code with your solution.

Reply:
Wonderful! Looks good!